File: packages/java/ygo74-agent-runtime/src/main/java/com/ygo74/agentruntime/domains/auth/JwtAuthenticator.java
package com.ygo74.agentruntime.domains.auth;

import java.util.Map;

public final class JwtAuthenticator {
    private JwtAuthenticator() {}

    public static Map<String, String> authenticate(String token, String userId) {
        if (token == null || token.isBlank()) {
            throw new IllegalArgumentException("missing token");
        }

        return Map.of("userId", userId, "authType", "jwt");
    }
}

File: packages/dotnet/Ygo74.AgentRuntime/Domains/Auth/JwtAuthenticator.cs
namespace Ygo74.AgentRuntime.Domains.Auth;

public static class JwtAuthenticator
{
    public static IDictionary<string, string> Authenticate(string token, string userId)
    {
        if (string.IsNullOrWhiteSpace(token))
        {
            throw new ArgumentException("missing token", nameof(token));
        }

        return new Dictionary<string, string>
        {
            ["userId"] = userId,
            ["authType"] = "jwt"
        };
    }
}
